Tests/SonarQube.TeamBuild.PreProcessor.Tests/Infrastructure/MockSonarQubeServer.cs

```diff
@@ -0,0 +1,119 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MockSonarQubeServer.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace SonarQube.TeamBuild.PreProcessor.Tests
+{
+    internal class MockSonarQubeServer : ISonarQubeServer
```

Comment:
Mock for the new interface: replaces the mocks for the property fetcher and rules generator.